mountpoint-s3-client/tests/put_object.rs

```diff
@@ -277,6 +279,65 @@ async fn test_put_review(pass_review: bool) {
     }
 }
 
+fn write_with_storage_class_test<F>(creator_fn: F, storage_class: &str)
```

Comment:
nit: these are tests on the s3 client, so maybe pick a name like `test_put_with_storage_class`